Declare command wallet support (#600)
software-mansion/protostar: website/docs/tutorials/06-deploying/01-cli.md

https://goerli.voyager.online/contract/0x038cc...
```

### Signing a declaration

As of cairo v0.10 you will be able to sign declare transactions. 
Our CLI will also provide 2 ways of providing the signature:

### 1. Using the default signer from starknet.py - [StarkCurveSigner](https://starknetpy.readthedocs.io/en/latest/signer.html#module-starknet_py.net.signer)

This way requires you to pass a private key (for signing) and account contract's address (to fetch the nonce). 

2 options are used for this:

> - `private-key-path` - a path to the file containing hex-encoded private key

Reviewer: how to obtain a keypair for signing? I think it would be nice to explain this in our docs or at least refer to source.

Author: To be honest i don't even know what to refer to here XD

Author: I can link to argentx, that's probably the most universal one, apart from that there is no clear demonstration on how to obtain the keys

Author: done

- `account-address` - your account contract's address (hex-encoded as well) on the appropriate network

Alternatively, if you prefer not to store private key in a file, we check for `ACCOUNT_PRIVATE_KEY` environment variable, and use it if it's available.   
It should be in the same hex-encoded format, like all the options above.

### 2. Using a custom signer class

> You can provide a custom signer class which conforms to [BaseSigner](https://starknetpy.readthedocs.io/en/latest/signer.html#starknet_py.net.signer.BaseSigner) Abstract Base Class specification. 

Reviewer: ```suggestion
You can provide a custom signer class which inherits from [BaseSigner](https://starknetpy.readthedocs.io/en/latest/signer.html#starknet_py.net.signer.BaseSigner) abstract class. 
```

> This way of signing requires you to write a class in python, which signs the transaction in a way that is suitable to you.

Reviewer: ```suggestion
This way of signing requires you to write a class in Python, which signs the transaction in a way that is suitable to you.
```

After writing such class, simply use `signer_class` argument in the CLI for `declare` command to use that class instead of the default one.
Usage of this way of signing is exclusive with the default signer strategy.

:::caution
The custom signer class must not take any arguments in the constructor, since we don't pass any args on instantiation.
:::

> You might wonder where you should put this class - we synchronize `PYTHONPATH` with project's `cairo_path`s, so you can put this near to your cairo contracts source code.

Reviewer: ```suggestion
The Python file containing this class can be put next to Cairo source code.
Protostar synchronizes `PYTHONPATH` with project's `cairo_path`.
```

> Modules that are dependencies of protostar (like starknet.py or cairo-lang) should be available for importing by default.

Reviewer: ```suggestion
Modules that are dependencies of Protostar (like `starknet_py` or `cairo-lang`) should be available for importing by default.
```

https://goerli.voyager.online/contract/0x038cc...
```

### Signing a declaration

> As of cairo v0.10 you will be able to sign declare transactions. 
> Our CLI will also provide 2 ways of providing the signature:

Reviewer: ```suggestion
Since Cairo v0.10 declare transactions can be signed.
Protostar offers two ways of providing the signature:
```
